compiler/codegen/OMRCodeGenerator.cpp

```diff
@@ -1142,7 +1138,7 @@ OMR::CodeGenerator::getNumberOfGlobalRegisters()
 #ifdef TR_HOST_S390
 uint16_t OMR::CodeGenerator::getNumberOfGlobalGPRs()
    {
-   if (self()->supportsHighWordFacility() && !self()->comp()->getOption(TR_DisableHighWordRA))
+   if (self()->supportsHighWordFacility())
```

Comment:
Pretty much all the code in the Z codegen checked for both whether we have high word support and whether the option was not enabled. Checking the option is rather expensive as it usually involves a few indirections. We changed this to only check the option when we set that we have high word support and if the option is specified we simply don't enable support.